### PR TITLE
Add design documents

### LINKED ISSUE #36: Add design documentation
Currently there are no documents in `jubakit` except for API reference.
I'm going to add design docs (contents copied from the presentation in Casual Talk event http://www.slideshare.net/kmaehashi/jubakit-63202847) to the docs merged in #33

### PR DESCRIPTION
**Pull request by kmaehashi**
This PR adds the design documents for jubakit; fixes #36.
Some contents are copied from http://www.slideshare.net/kmaehashi/jubakit-63202847

### CODE REVIEW

File: doc/source/architecture/dataset.rst [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+  from jubakit.classifier import Schema, Dataset
+
+  loader = ...
+  schema = Schema( ... )
+
+  dataset = Dataset(loader, schema)
+
+Some Services provides additional ways to construct Dataset.
+For example, :py:class:`jubakit.classifier.Dataset` provides :py:func:`from_array <jubakit.classifier.Dataset.from_array>` and :py:func:`from_matrix <jubakit.classifier.Dataset.from_matrix>` which are convenient when using datasets generated by scikit-learn.
+Other than noted, features mentioned in this section are implemented in :py:class:`jubakit.base.BaseDatset`, which is a base Dataset class for all Services.
```
Comment (shiodat) [inline doc/source/architecture/dataset.rst]: `jubakit.base.Datset` -> `jubakit.base.Dataset`

File: doc/source/guide/loader_develop.rst [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+
+  def preprocess(self, ent):
+    return ent
+
+Although not mandatory, the output of ``preprocess`` method should be a flat dict-like object, i.e., values should not be objects.
+
+Transformation
+~~~~~~~~~~~~~~
+
+For example, if you want to process JSONL files (files that contain one JSON record per line), you can create a class that inherits from :py:class:`LineBasedFileLoader <jubakit.loader.core.LineBasedFileLoader>` which loads a single text file and emits record for each line, and implement ``preproces`` method as follows:
```
Comment (shiodat) [inline doc/source/guide/loader_develop.rst]: ``preproces`` -> ``preprocess``

**Comment by shiodat** [comment]
LGTM :+1:  
Please fix some typos.

**Comment by kmaehashi** [comment]
Thanks! fixed.

**Comment by shiodat** [comment]
:+1: Thanks!

**Pull request merged**